### ISSUE TITLE
Example of using using authorization controls with the services?

### ISSUE THREAD

**Issue by jeffkole** [yahoo/fluxible-plugin-fetchr#23]
Do you have any examples of requiring authorization for the Fetchr services?  I can see two ways of going about it and wouldn't mind some thoughts from the authors.

Put authorization calls into the body of the protected functions:

```
todoDataService = {
  read: function (request, resource, params, config, callback) {
    if (!request.user) {
      return callback({ notAuthenticated: true });
    }
    if (request.user && !request.user.canReadTodos()) {
      return callback({ notAuthorized: true });
    }
    return ... ; // the actual data
  }
}
```

Add authentication and authorization middleware before registering the Fetchr middleware:

```
app.use(fetchr.getXhrPath(), authMiddleware);
app.use(fetchr.getXhrPath(), fetchr.getMiddleware());
```

In this second case, `authMiddleware` would need to be aware of the resources that could be served by Fetcher.

**Comment by jhwang09** [comment]
Have you made this work? this seems to be the right approach, since you're using express middleware

**Comment by jeffkole** [comment]
I only have two fetchers right now and have gone with the first approach so far.  While it requires putting auth code in more places, it means that the decision to protect an endpoint is right with the code that it is protecting.  In the second approach, the `authMiddleware` would need to be intimately familiar with the fetchers that are in use.  In some authorization cases, you need to read some data from the database before knowing that the user is authorized to act.  That seemed like too much to generalize quickly in a single `authMiddleware`.

**Comment by jhwang09** [comment]
Yes, seems like the first approach is the right way to do it. It's like the express bodyparser() where it parse data for you beforehand so you can use that data more efficiently. I think I would go with the first approach. (I am actually figuring this out too)

**Comment by tleunen** [comment]
I'd love to have something similar to the `next()` function in express. This would allow us to chain the service and easily add authorization to our services.
